src/securejoin.rs

```diff
@@ -516,10 +515,11 @@ pub(crate) async fn handle_securejoin_handshake(
                 // no error, just aborted somehow or a mail from another handshake
                 return Ok(HandshakeMessage::Ignore);
             }
-            let scanned_fingerprint_of_alice = get_qr_attr!(context, fingerprint).to_string();
+            let scanned_fingerprint_of_alice: Fingerprint =
+                get_qr_attr!(context, fingerprint).clone();
```

Comment:
The clone is needed to turn the ref from `get_qr_attr!()` into an owned fingerprint.  This is the same as the previous `.to_string()` which made it owned.  Both these are sadly needed because the macro call contains an `.await` and that makes the references not live long enough.